runtime/runtime.go

```diff
@@ -150,6 +150,15 @@ func (r *interpreterRuntime) ExecuteScript(
 	}
 	epSignature := invokableType.InvocationFunctionType()
 
+	storableResults := map[*sema.Member]bool{}
+	if !epSignature.ReturnTypeAnnotation.Type.Equal(&sema.VoidType{}) {
```

Comment:
Could just be a Go type check to avoid the allocation:

```suggestion
	if _, isVoid := epSignature.ReturnTypeAnnotation.Type.(*sema.VoidType); !isVoid {
```